CommonTools/PileupAlgos/src/PuppiContainer.cc

```diff
@@ -322,8 +324,13 @@ std::vector<double> const &PuppiContainer::puppiWeights() {
     //Basic Cuts
     if (pWeight * fPFParticles[i0].pt() < fPuppiAlgo[pPupId].neutralPt(fNPV) && rParticle.id == 0)
       pWeight = 0;  //threshold cut on the neutral Pt
-    if ((fPtMax > 0) && (rParticle.id == 0))
-      pWeight = min(max(pWeight, fPFParticles[i0].pt() / fPtMax), 1.);
+    // Protect high pT photons (important for gamma to hadronic recoil balance)
+    if ((fPtMaxPhotons > 0) && (rParticle.pdgId == 22) && (std::abs(fPFParticles[i0].eta()) < fEtaMaxPhotons) &&
+        (fPFParticles[i0].pt() > fPtMaxPhotons))
+      pWeight = 1.;
+    // Protect high pT neutrals
+    if ((fPtMaxNeutrals > 0) && (rParticle.id == 0))
```

Comment:
it may be more clear to say here ```else if ```.
It's a bit more trivial than to figure out that the pWeight set to 1 above will not be changed if it inters this condition.

Reply:
considering the comment in https://github.com/cms-sw/cmssw/pull/27929#issuecomment-535917262
I'm not 100% sure now that this if -> else if is logically identical.

Reply:
the "if else" should recover the same logic that PuppiPhoton had (so without the else it was actually wrong). However, all of this not should no affect puppi (jets), but only puppiNoLeptons